tests/utils.py

```diff
@@ -865,6 +865,16 @@ def run_emulated(simulation: td.Simulation, path=None, **kwargs) -> td.Simulatio
     """Emulates a simulation run."""
     from scipy.ndimage.filters import gaussian_filter
 
+    def get_spatial_coords(grid, monitor, field_name):
+        spatial_coords = grid.boundaries if monitor.colocate else grid[field_name]
+        spatial_coords_dict = spatial_coords.dict()
+
+        return {
+            dim: (coords[:-1] if len(coords) > 1 else coords)
```

Comment:
I think the issue causing the test to fail is here. In my test, it is missing one coordinate because here it is dropped. I am sure you had a reason to drop it, but maybe the fix is to drop it if `colocate=True`, but keep it when `colocate=False`.